T03 category menu bar

Blyatman0692 opened the pull request:
Added category menu bar that maps each category to correct page section.
** needs additional styling
** smooth scrolling animation is not working

Blyatman0692 committed:
Nav Menu with items extracted from categories.json

Blyatman0692 committed:
Functional category menu with correct category mapping.
**Smooth Scrolling is NOT working

Crazypandaz committed:
-Updating html tags in Create, Read, and Update page file to improve font size and layout slightly.

nurselaine commented on ContosoCrafts.WebSite/Services/JsonFileCategoryService.cs:
```diff
@@ -0,0 +1,32 @@
+using Microsoft.AspNetCore.Hosting;
+using ContosoCrafts.WebSite.Models;
+
+namespace ContosoCrafts.WebSite.Services
```
There should already be a method inside of JsonFileMenuService to deserialize the category json file - If you would like to use refactor the code then we can reuse what is in JsonFileMenuService or if you feel it's more appropriate you can choose to remove the GetCategories method from inside of the JsonFileMenuService

nurselaine commented on ContosoCrafts.WebSite/Models/Category.cs:
```diff
@@ -0,0 +1,9 @@
+﻿using System;
+namespace ContosoCrafts.WebSite.Models
+{
+	public class Category
```
There is a ProductCategory class inside of the Menu models created that represents the categories json file data. Let's decide on whether we want to refactor it into its own model or if it belongs within menu items model.

Crazypandaz committed:
-Added Review Link near top of product popup.

Crazypandaz committed:
-Added Review page file for showing selected product along with its associated reviews. Comes with capability to submit review and rating submission at the bottom.

Crazypandaz committed:
-Added Review page model file to handle request logic behind Review page.

Crazypandaz committed:
-Added Review model file to represent review object in review.json.

Crazypandaz committed:
-added css tags for rating star behvaiors.
-added css tags to layout review webpage as three separate columns: review, vertical line, and product.

Crazypandaz committed:
-Added JsonFileReviewService to Startup.cs.

Crazypandaz committed:
-Added mock json data for reviews.

Crazypandaz committed:
-Added JsonFileReviewService to interface between review page and review.json.

Crazypandaz committed:
-Modified AddRating() within JsonFileMenuService to use 'Id' variable instead of 'Name' variable in sequence filter.

Crazypandaz committed:
-Modified MenuList razor component to remove the obsolete star rating elements in popup footer.

Crazypandaz committed:
-Added model validation rules to Review model file.

macnoblebk committed:
Merge pull request #15 from nurselaine/T15-Posting-Comment/Review-On-Items

T15 posting comment/review on items

nurselaine committed:
Merging service conflicts
